compiler/rustc_parse/src/parser/path.rs

```diff
@@ -383,6 +383,16 @@ impl<'a> Parser<'a> {
 
                 PathSegment { ident, args: Some(args), id: ast::DUMMY_NODE_ID }
             } else {
+                if style == PathStyle::Pat
+                    && self.check_noexpect(&token::Lt)
+                    && self.look_ahead(1, |t| t.can_begin_type())
+                {
+                    self.sess.emit_err(GenericArgsInPatRequireTurbofishSyntax {
+                        span: self.token.span,
+                        suggest_turbofish: self.token.span.shrink_to_lo(),
+                    });
+                }
```

Comment:
Note that by emitting the error but continuing the parse state as is, we are producing multiple spurious parse errors after this. Ideally, we would generate a snapshot, attempt to continue parsing as if this was a path, and if so continue from there.

Reply:
How about returning the error directly? I have no idea when to generate and restore a snapshot. IMO, we do not need to continue parsing when encountering this error. But this would make the suggestion for anonymous parameters disappear.